### DISCUSSION TITLE
CertoraRun: You dont have have permission

### DISCUSSION THREAD

**Discussion by 0x0shephard** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#13] [Q&A] [answered]
<img width="670" alt="xkskxms" src="https://github.com/Cyfrin/assembly-evm-opcodes-and-formal-verification-course/assets/67645307/a7aadba0-3b11-4de9-a0c4-7a51c41193a2">

I getting this issue with certora key while running certoraRun that I dont have the permission to connect to the server

**Comment by EngrPips** [comment]
Hello @jonraza15, Did you provide a valid Certora `api-key` to the process?

**Reply by 0x0shephard** [reply]
yes and echoed it too

**Reply by EngrPips** [reply]
Honestly, I will advise that you check that you have copied correctly your `api-key`  and also ensure that when you tried exporting it to the process environment, you didn't add any whitespace at the end of the `api-key.`

**Reply by 0x0shephard** [reply]
did all of those still getting the issue

**Reply by EngrPips** [reply]
I am not sure what could be wrong but I know for sure computers don't lie. You are surely doing something wrong. Maybe take a break and come back to it with a clear head.

**Reply by 0x0shephard** [reply, marked as answer]
got help from certora discord, first letter of my key was uppercase, which should have been lowercase, thanks nonetheless, issue is resolved

**Reply by EngrPips** [reply]
You claimed you correctly copied your key, so did Certora give you the wrong key? I am just curious.

**Reply by 0x0shephard** [reply]
yes

**Reply by EngrPips** [reply]
Wow, Interesting!
